synapse/util/caches/descriptors.py

```diff
@@ -286,7 +271,7 @@ def __init__(self, orig, cached_method_name, list_name, num_args=None):
 
     def __get__(self, obj, objtype=None):
         cached_method = getattr(obj, self.cached_method_name)
-        cache = cached_method.cache
+        cache = cached_method.cache  # type: DeferredCache[Tuple, Any]
```

Comment:
Should this also be `CacheKey`, not `Tuple`?

```suggestion
        cache = cached_method.cache  # type: DeferredCache[CacheKey, Any]
```

Reply:
yes, thanks.